cmd/pro/provider/list/templateoptions.go

```diff
@@ -116,18 +116,37 @@ func parametersToOptions(parameters []storagev1.AppParameter) map[string]*types.
 		for _, opt := range parameter.Options {
 			paramOpts = append(paramOpts, types.OptionEnum{Value: opt})
 		}
+
 		options[optionName] = &types.Option{
 			DisplayName: displayName,
 			Description: parameter.Description,
 			Required:    parameter.Required,
 			Enum:        paramOpts,
 			Default:     parameter.DefaultValue,
 			Mutable:     true,
+			Type:        getOptionType(parameter),
```

Comment:
what does this do? Why not just use `parameter.Type`?

Reply:
The `storagev1.AppParameter.Type` isn't a 1:1 mapping necessarily to `types.Option`, I wanted to have a clear place where we translate from one to the other, even though right now they are _mostly_ similar (Option supports `datetime` which appparam doesnt, but appparam supports `password` which option doesn't 😅 )

Reply:
Oh I see now, so although the parameterToOptionTypeMap has 1 to 1 mappings, it will allow us to use an empty string for unsupported options like "datetime"